src/Symfony/Component/Runtime/Runner/Middleware/MiddlewareFactory.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Runtime\Runner\Middleware;
+
+/**
+ * @author Sascha Heilmeier <sascha.heilmeier@netlogix.de>
+ */
+readonly class MiddlewareFactory
```

Comment:
non-internal non-final classes must not be `readonly`, as the BC rules would forbid us to ever remove that keyword in the future due to the impact this keyword has on inheritance.

Reply:
The factory is now final and internal.